internal/service/ecs/task_definition_test.go

```diff
@@ -167,6 +167,62 @@ func TestAccECSTaskDefinition_withDockerVolumeMinimal(t *testing.T) {
 	})
 }
 
+func TestAccECSTaskDefinition_withRuntimePlatform(t *testing.T) {
+	var def ecs.TaskDefinition
+
+	tdName := sdkacctest.RandomWithPrefix("tf-acc-td-with-runtime-platform")
+	resourceName := "aws_ecs_task_definition.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, ecs.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckTaskDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTaskDefinitionWithRuntimePlatformMinimalConfig(tdName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTaskDefinitionExists(resourceName, &def),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateIdFunc: testAccTaskDefinitionImportStateIdFunc(resourceName),
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccECSTaskDefinition_Fargate_withRuntimePlatform(t *testing.T) {
+	var def ecs.TaskDefinition
+
+	tdName := sdkacctest.RandomWithPrefix("tf-acc-td-with-runtime-platform")
+	resourceName := "aws_ecs_task_definition.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, ecs.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckTaskDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFargateTaskDefinitionWithRuntimePlatformMinimalConfig(tdName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTaskDefinitionExists(resourceName, &def),
```

Comment:
same as above

Reply:
K; added them in but I'm not sure I get the point in unit testing a string constant hcl record for a string?
Is it supposed to be testing the hcls parsed properly? Seems an odd thing for an integration test vs unit 🤷

Reply:
well, for starters this is not really a unit test. this tests the above code and integration with AWS E2E. but this is to check expected result equals the actual one.

Reply:
Yeah tbh I was looking at the method names rather than digging into what each one was doing; it "looked" like it was just checking the string was in the hcl (i.e. checking that what was typed was typed which seemed pointless) 
But I see its checking the hcl as a map essentially checking the HCL will render (i pressume thats the intent)

The unit-test comment was solely about that section as its not hitting the API its just checking the terraform config format. Slows life down when you have to wait 14 seconds and you're blindly using someone elses code base filled with hundreds of helper methods you've never seen before 😄